features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapListener.java

```diff
@@ -102,15 +108,49 @@ public void trapError(int error, String msg) {
     }
 
     public void start() {
-        m_twinSubscription = m_twinSubscriber.subscribe(TrapListenerConfig.TWIN_KEY, TrapListenerConfig.class, (config) ->  {
+        if (m_twinSubscriber != null) {
+            subscribe();
+        } else {
+            new Timer().schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    delayedConfigurationCheck();
+                }
+            }, 60 * 1000); // wait 60 seconds for connection from core
+        }
+    }
+
+    private void delayedConfigurationCheck() {
+        synchronized (m_configured) {
+            if (!m_configured.get()) {
+                LOG.warn("No trap configuration received from core, using default settings");
+                this.open(new TrapListenerConfig());
+            }
+        }
+    }
+
+    public void subscribe() {
+        m_twinSubscription = m_twinSubscriber.subscribe(TrapListenerConfig.TWIN_KEY, TrapListenerConfig.class, (config) -> {
             try (Logging.MDCCloseable mdc = Logging.withPrefixCloseable(Trapd.LOG4J_CATEGORY)) {
                 LOG.info("Got listener config update - reloading");
-                this.close();
-                this.open(config);
+                synchronized(m_configured) {
```

Comment:
since m_configured is already atomic boolean, do you still need synchronization here ?

Reply:
I think so. For the TimerTask, checking the boolean value and then opening the trap listener need to be atomic with respect to this code block. My thinking is that if the TimerTask decides to start opening the listener with a default config, we want that to finish completely before this is attempted, if they happen to occur at the same time. m_configured doesn't need to be an AtomicBoolean, I just started with it that way and I wanted an object to synchronize around. I'll switch it to a standard Boolean object